### ISSUE TITLE
Use <optgroup> and <option> element for optionGroups in Select and MultiSelect

### ISSUE THREAD

**Issue by ewillebrands** [primefaces/primevue#8204]
### Describe the bug

An option group within a select or multiselect component is currently rendered as a regular option in the DOM. This is semantically incorrect, and screenreaders will read the group titles as valid options for the select. They should be defined as optgroup instead, and have the child <option> elements hierarchically structured within them. https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/optgroup
This is a dedicated HTML element for this. Using this correctly will enhance accessibility.

Or alternatively, if this is impossible due to the more complex structure of the component, aria-roles can be used: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/listbox_role
This also requires the options to be nested as children in the groups.

### Reason for not contributing a PR

- [x] Lack of time
- [ ] Unsure how to implement the fix/feature
- [ ] Difficulty understanding the codebase
- [ ] Other

### Reproducer

https://stackblitz.com/edit/primevue-4-vite-issue-template-5domgbvm?file=src%2FApp.vue

### Environment

This is already a problem on your own environment https://primevue.org/select/#group.

### Vue version

3.3.0

### PrimeVue version

4.4.1

### Steps to reproduce the behavior

1. Go to https://primevue.org/select/#group
2. Open the select, and inspect the title of a group
3. In the DOM, this is just a list item of the same hierarchical level as all options (which are also list elements)

### Expected behavior

I expect the options to be nested inside the optiongroup as children of them, having the appropriate element tags or aria-roles.

**Comment by tugcekucukoglu** [comment]
This issue will be fixed in version 5.0.2. Thank you.
